Zattoo/Fastfile

```diff
@@ -0,0 +1,183 @@
+
+desc "Run Unit tests"
+lane :unit_tests do
+  cocoapods(repo_update: true)
+  scan(scheme: "ZattooTests")
+  post_to_slack(message: "*Unit Tests*: All tests successfully passed", success: true)
+end
+
+desc "Run UI Tests"
+lane :ui_tests do
+  cocoapods(repo_update: true)
+  scan(scheme: "ZattooUITests")
+  post_to_slack(message: "*UI Tests*: All tests successfully passed", success: true)
+end
+
+desc "Check style and conventions"
+lane :lint do
+  swiftlint(strict: true)
+end
+
+#### Releases ####
+
+desc "Release build to TestFlight"
+lane :release do
+  ensure_git_status_clean
+  increment_build(app_identifier: Zattoo::AppIdentifier::Production) 
+  gym
+  pilot
+  post_to_slack(message: "New *Zattoo* available in TestFlight", success: true)
+  reset_git_repo
+end
+
+desc "Release a Preview version to TestFlight"
+lane :release_preview do
+  ensure_git_status_clean
+  setup_preview
+  increment_build(app_identifier: Zattoo::AppIdentifier::Preview) 
+  setup_preview_provisioning_profiles
+  gym
+  pilot
+  post_to_slack(message: "New *Zattoo Preview* available in TestFlight", success: true)
+  reset_git_repo
+end
+
+desc "Release a new version for boxes on the Apple Stores"
+lane :release_applestore do
+  ensure_git_status_clean
+  setup_applestore
+  increment_build(app_identifier: Zattoo::AppIdentifier::AppleStore)
+  setup_applestore_provisioning_profiles
+  gym
+  pilot
+  post_to_slack(message: "New *Zattoo AppleStore* available in TestFlight", success: true)
+  reset_git_repo
+end
+
+desc "Generate screenshots"
+lane :screenshots do
+  snapshot
+end
+
+#### Error management ####
+
+error do |lane, exception|
+  case lane
+  when :lint then post_to_slack(message: "Linter finished with errors", success: false)
+  when :unit_tests then post_to_slack(message: "UnitTests finished with errors", success: false)
+  when :ui_tests then post_to_slack(message: "UITests finished with errors", success: false)
+  else post_to_slack(message: "Finished with errors", success: false)
+  end
+end
+
+#### Private ####
+
+private_lane :post_to_slack do |options|
+  if ENV["SLACK_URL"]
+    message = options[:message] 
+    success = options[:success]
+    slack(message: ":tv: tvOS: " + message + (success ? " :beers:" : " :bomb:"), success: success)
+  end
+end
+
+#### Configuration for ZattooPreview ####
+
+private_lane :setup_preview do
+  setup_preview_icon
+  setup_preview_display_name
+  setup_preview_app_identifier
+end
+
+private_lane :setup_preview_icon do
+  sh "helpers/setupPreviewIcon.sh"
+end
+
+private_lane :setup_preview_app_identifier do
+  update_app_identifier(plist_path: "Zattoo/Resources/Info.plist", app_identifier: Zattoo::AppIdentifier::Preview)
+  update_app_identifier(plist_path: "TopShelfExtension/Resources/Info.plist", app_identifier: Zattoo::TopShelfAppIdentifier::Preview)
+end
+
+private_lane :setup_preview_display_name do
+  update_info_plist(plist_path: "Zattoo/Resources/Info.plist", display_name: Zattoo::DisplayName::Preview)
```

Comment:
@pvinis those values are defined in a separate ruby file, imported on top of the Fastfile with:

```
require './helpers/zattoo.rb'
```

This ruby file has a format similar to the one bellow:

```
class MyCompany

   MyVariable = "Hey dude"

   class MySecondVariable
      Production = "Value For Production"
      QA = "Value for QA"
   end

end
```

So you could use `MyCompany::MyVariable` or `MyCompany::MySecondVariable::Production` in your Fastfile, avoiding the use of literal strings everywhere.